classes/helper.php

```diff
@@ -84,4 +85,28 @@ public static function is_https() {
 
 		return strncmp( $url, 'https:', 6 ) === 0;
 	}
+
+
+	/**
+	 * Extends the error logging with request and response data.
+	 *
+	 * @param Exception|Error $e The exception to log.
+	 * @param array           $arr The array to extend with additional data.
+	 * @return array The same or extended array if ZaverError.
+	 */
+	public static function extra_logging( $e, $arr ) {
```

Comment:
I would suggest changing the name of this method. `extra_logging` does not fully describe what this does. And i would suggest calling it something along the line of `add_request_log_context` or similar, just to avoid any confusion in the future.